packages/puppeteer-core/src/bidi/core/Realm.ts

```diff
@@ -0,0 +1,256 @@
+import type * as Bidi from 'chromium-bidi/lib/cjs/protocol/protocol.js';
+
+import {EventEmitter} from '../../common/EventEmitter.js';
+
+import type {BrowsingContext} from './BrowsingContext.js';
+import type {Connection} from './Connection.js';
+
+export type CallFunctionOptions = Omit<
+  Bidi.Script.CallFunctionParameters,
+  'functionDeclaration' | 'awaitPromise' | 'target'
+>;
+
+export type EvaluateOptions = Omit<
+  Bidi.Script.EvaluateParameters,
+  'expression' | 'awaitPromise' | 'target'
+>;
+
+export abstract class Realm extends EventEmitter<{
+  /** Emitted when the realm is destroyed. */
+  destroyed: void;
+  /** Emitted when a dedicated worker is created in the realm. */
+  worker: DedicatedWorkerRealm;
+  /** Emitted when a shared worker is created in the realm. */
+  sharedworker: SharedWorkerRealm;
+}> {
+  readonly id: string;
+  readonly origin: string;
+  protected constructor(id: string, origin: string) {
+    super();
+    this.id = id;
+    this.origin = origin;
+  }
+
+  protected initialize(): void {
+    this.connection.on('script.realmDestroyed', info => {
+      if (info.realm === this.id) {
+        this.emit('destroyed', undefined);
+      }
+    });
+  }
+
+  protected abstract get connection(): Connection;
+
+  protected get target(): Bidi.Script.Target {
```

Comment:
Use The Realm Target type

Reply:
The `WindowRealm` uses `script.ContextTarget` instead of `script.RealmTarget` for its `Target` type.